src/Microsoft.AspNetCore.Cors/Infrastructure/CorsPolicyExtensions.cs

```diff
@@ -16,7 +16,9 @@ public static bool IsOriginAnAllowedSubdomain(this CorsPolicy policy, string ori
             {
                 return true;
             }
-            var originUri = new Uri(origin, UriKind.Absolute);
+
+            Uri.TryCreate(origin, UriKind.Absolute, out var originUri);
```

Comment:
And if it fails? Does UriHelpers.IsSubdomainOf accept null?

Reply:
Yeah but it returns false here https://github.com/aspnet/CORS/blob/dev/src/Microsoft.AspNetCore.Cors/Infrastructure/UriHelpers.cs#L12

Reply:
https://github.com/aspnet/CORS/blob/c873fc55f9fe8ff5d0de2e54cd763cd4a776c42b/src/Microsoft.AspNetCore.Cors/Infrastructure/UriHelpers.cs#L13 will null ref if you ever get there.

Reply:
Oh my tests were short circuiting before even going there. Let me look into it. Thanks